File: client/src/types/address.ts
import { BaseModelExtension } from "./baseModelExtension";

export interface Address extends BaseModelExtension {
    careOf: string | null;
    postCode: string | null;
    city: string | null;
    languague: string | null;
    country: string | null;
    type: AddressType;
}

export enum AddressType {
    StreetAddress = 1,
    PostalAdress = 2
}

File: server/src/Models/Address.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace server.src.Models
{
    public class Address : BaseModelExtension
    {
        public string? CareOf { get; set; }
        public string? PostCode { get; set; }
        public string? City { get; set; }
        public string? Languague { get; set; }
        public string? Country { get; set; }
        public AddressType Type { get; set; }
    }

    public enum AddressType
    {
        StreetAddress = 1,
        PostalAdress = 2
    }
}